src/ReachSets/DiscretePost/LazyDiscretePost.jl

```diff
@@ -3,40 +3,45 @@
 # ==============================================================================
 
 import LazySets.use_precise_ρ
+import Reachability.check_aliases_and_add_default_value!
 
-struct LazyTextbookDiscretePost <: DiscretePost
+struct LazyDiscretePost <: DiscretePost
     options::Options
-end
 
-# default options for the LazyTextbookDiscretePost discrete post operator
-function LazyTextbookDiscretePost()
-    defaults = Options()
-    setindex!(defaults, Hyperrectangle, :overapproximation)
-    setindex!(defaults, false, :check_invariant_intersection)
-    setindex!(defaults, false, :lazy_R⋂I)
-    setindex!(defaults, true, :lazy_R⋂G)
-    setindex!(defaults, true, :lazy_A⌜R⋂G⌟⋂I)
-    return LazyTextbookDiscretePost(defaults)
+    function LazyDiscretePost(𝑂::Options)
+        𝑂copy = copy(𝑂)
+        # TODO: pass 𝑂 directly?
+        check_aliases_and_add_default_value!(𝑂copy.dict, 𝑂.dict, [:check_invariant_intersection], false)
+        check_aliases_and_add_default_value!(𝑂copy.dict, 𝑂.dict, [:overapproximation], Hyperrectangle)
+        check_aliases_and_add_default_value!(𝑂copy.dict, 𝑂.dict, [:lazy_R⋂I], false)
+        check_aliases_and_add_default_value!(𝑂copy.dict, 𝑂.dict, [:lazy_R⋂G], true)
+        check_aliases_and_add_default_value!(𝑂copy.dict, 𝑂.dict, [:lazy_A⌜R⋂G⌟⋂I], true)
+        return new(𝑂)
+    end
 end
 
-function init(op::LazyTextbookDiscretePost, system, options_input)
-    options_input.dict[:n] = statedim(system, 1)
+# convenience constructor from pairs of symbols
+LazyDiscretePost(𝑂::Pair{Symbol,<:Any}...) = LazyDiscretePost(Options(Dict{Symbol,Any}(𝑂)))
+
+# default options for the LazyDiscretePost discrete post operator
+LazyDiscretePost() = LazyDiscretePost(Options())
+
+init(𝒟::LazyDiscretePost, 𝒮::AbstractSystem, 𝑂::Options) = init!(𝒟, 𝒮, copy(𝑂))
```

Comment:
Hm, why the new symbols? In particular, 𝒟 can be misread for a `DiscreteSystem`.
Should we use them everywhere to be consistent?

We should define this function for an arbitrary `DiscretePost` type in the interface file.

Reply:
> Hm, why the new symbols?

i was looking for some notation for the post operator. perhaps some "P" or "P_d" for discrete post and "P_c" for cont post?

Reply:
It's difficult... Except for `solve.jl`, you only work with one operator, so you do not need to explicitly say which type of operator you have. Meaning that just `P` would work.